### PR TITLE
Adding support for ANY_KEYS_MATCH udf in presto

### PR DESCRIPTION
**Pull request by jainavi17**
Test plan

Added unit tests.
Build successfully using the following terminal command
- `./mvnw clean install -Dtest=TestAnyKeysMatchFunction -fn -pl presto-main`

```
== RELEASE NOTES ==

General Changes
* Add function :func:`any_keys_match`
```

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    @Test
+    public void testEmpty()
+    {
+        assertFunction("ANY_KEY_MATCH(MAP(ARRAY[], ARRAY[]), (x) -> x > 0)", BOOLEAN, false);
+    }
+
+    @Test
+    public void testNull()
+    {
+        assertFunction("ANY_KEY_MATCH(NULL, (x) -> x LIKE '%ab%' )", BOOLEAN, null);
```
Comment (feilong-liu) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java:55]: Can you also add a test case where the element in the map has key to be null?

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                BOOLEAN,
+                false);
+    }
+
+    @Test
+    public void testError()
+    {
+        assertInvalidFunction(
+                "MAP_TOP_N(MAP(ARRAY[ROW('x', 1), ROW('y', 2)], ARRAY[1, 2]), (x) -> x[2] LIKE '%ab%')",
+                SemanticErrorCode.FUNCTION_NOT_FOUND);
```
Comment (feilong-liu) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java:80]: why this case throws error of FUNCTION_NOT_FOUND?

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java (hunk 1 above)
Comment (feilong-liu) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java:55]: And also cases where predicate function returns null for non null keys.

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java (hunk 2 above)
Comment (jainavi17) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java:80]: Cause the left hand side is integer which is being compared with Varchar

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java (hunk 1 above)
Comment (jainavi17) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java:55]: Null keys aren't allowed.

Let me add test case when predicate function returns null

Comment (kaikalur) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java:55]: Still makes sense to test that it doesn;'t crash and/or gives unexpected answers.

Comment (jainavi17) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java:55]: I've updated the test cases with function returning null

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    public void testEmpty()
+    {
+        assertFunction("ANY_KEY_MATCH(MAP(ARRAY[], ARRAY[]), (x) -> x > 0)", BOOLEAN, false);
+    }
+
+    @Test
+    public void testNull()
+    {
+        assertFunction("ANY_KEY_MATCH(NULL, (x) -> x LIKE '%ab%' )", BOOLEAN, null);
+        assertFunction("ANY_KEY_MATCH(MAP(ARRAY['x', 'y'], ARRAY[1, 2]), (x) -> CAST(NULL AS BOOLEAN))", BOOLEAN, null);
```
Comment (feilong-liu) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java:56]: > Still makes sense to test that it doesn;'t crash and/or gives unexpected answers. Simple example: x->if(x>0, true)

Can you also add a case where predicate returns true for some elements and null for other elements, using predicates like suggested above.

**Comment by feilong-liu** [comment]
For the release note, your change is not specifically related to hive, so do not need to include the hive section.
Also the description of the udf should be a sub bullet point, something like:

- add xxx
  - xxxx

**Comment by feilong-liu** [comment]
> For the release note, your change is not specifically related to hive, so do not need to include the hive section. Also the description of the udf should be a sub bullet point, something like:
> 
> * add xxx
>   
>   * xxxx

And the same for the other two PRs

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java (hunk 3 above)
Comment (feilong-liu) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java:56]: Can you address the above comments?

Comment (jainavi17) [reply presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestAnyKeyMatchFunction.java:52]: > Can you address the above comments?

Done!

**Comment by feilong-liu** [comment]
Check the maven checks failures, you have code style violations.

**Review by feilong-liu — APPROVED** [review]

**Review by rschlussel — COMMENTED** [review]
Can you rename the function  `any_key_matches` or `any_keys_match`?  just to make it better English.

**Comment by rschlussel** [comment]
for all the release notes for added functions, you can just add a reference to the documentation instead of rewriting the docs as the release note.  You can add a link to the documentation using the :func:`function_name`  syntax

e.g. 
```
* Add function :func:`any_keys_match`
```

**Review by rschlussel — APPROVED** [review]
